File: my_py_pkg/my_py_pkg/hw_status_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from my_robot_interfaces.msg import HardwareStatus

class HardwareStatusPublisherNode(Node): 
    def __init__(self):
        super().__init__("hardware_status_publisher")

        # Create publisher obj
        self.publisher_ = self.create_publisher(HardwareStatus, "hardware_status", 10) # type, topic name, queue size

        # add timer
        self.timer_ = self.create_timer(0.5, self.publish_hardware_status)

        self.get_logger().info("Hardware status Publisher has been started")

    def publish_hardware_status(self):
        # Create msg obj
        msg = HardwareStatus()
        # Add custom data
        msg.temperature = 25
        msg.are_motors_ready = True
        msg.debug_message = "Nothing special"

        # Output the data
        self.publisher_.publish(msg)
        self.get_logger().info("Message published!")


def main(args=None):
    rclpy.init(args=args)
    node = HardwareStatusPublisherNode() 
    rclpy.spin(node)
    rclpy.shutdown()


if __name__=="__main__":
    main()

File: my_cpp_pkg/src/hw_status_publisher.cpp
#include "rclcpp/rclcpp.hpp"

#include "my_robot_interfaces/msg/hardware_status.hpp"

class HardwareStatusPublisherNode : public rclcpp::Node // MODIFY NAME
{
public:
    HardwareStatusPublisherNode() : Node("hardware_status_publisher") // MODIFY NAME
    {
        publisher_ = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>("hardware_status",10);
        timer_ = this->create_wall_timer(std::chrono::milliseconds(500), std::bind(&HardwareStatusPublisherNode::publishHardwareStatus, this));
        RCLCPP_INFO(this->get_logger(), "Hardware status publisher has been started.");
    }

    void publishHardwareStatus()
    {
        auto msg = my_robot_interfaces::msg::HardwareStatus();
        msg.temperature = 40;
        msg.are_motors_ready = true;
        msg.debug_message = "Nothing new";
        publisher_->publish(msg);
    }


    rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;

private:

};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<HardwareStatusPublisherNode>();   // MODIFY NAME
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}